core/services/vrf/integration_v2_test.go

```diff
@@ -381,6 +385,94 @@ func registerProvingKeyHelper(t *testing.T, uni coordinatorV2Universe, vrfkey vr
 	uni.backend.Commit()
 }
 
+func TestExternalOwnerConsumerExample(t *testing.T) {
+	owner := newIdentity(t)
+	random := newIdentity(t)
+	genesisData := core.GenesisAlloc{
+		owner.From: {Balance: assets.Ether(10)},
+	}
+	backend := cltest.NewSimulatedBackend(t, genesisData, ethconfig.Defaults.Miner.GasCeil)
+	linkAddress, _, linkContract, err := link_token_interface.DeployLinkToken(
+		owner, backend)
+	require.NoError(t, err)
+	backend.Commit()
+	coordinatorAddress, _, coordinator, err :=
+		vrf_coordinator_v2.DeployVRFCoordinatorV2(
+			owner, backend, linkAddress, common.Address{}, common.Address{})
+	require.NoError(t, err)
+	_, err = coordinator.SetConfig(owner, uint16(1), uint32(10000), 1, 1, big.NewInt(10), vrf_coordinator_v2.VRFCoordinatorV2FeeConfig{
+		0, 0, 0, 0, 0, big.NewInt(0), big.NewInt(0), big.NewInt(0), big.NewInt(0),
+	})
+	require.NoError(t, err)
+	backend.Commit()
+	consumerAddress, _, consumer, err := vrf_external_sub_owner_example.DeployVRFExternalSubOwnerExample(owner, backend, coordinatorAddress, linkAddress, 1, 1, 1, [32]byte{})
+	require.NoError(t, err)
+	backend.Commit()
+	_, err = linkContract.Transfer(owner, consumerAddress, assets.Ether(2))
+	require.NoError(t, err)
+	backend.Commit()
+	AssertLinkBalances(t, linkContract, []common.Address{owner.From, consumerAddress}, []*big.Int{assets.Ether(999_999_998), assets.Ether(2)})
+
+	// Create sub, fund it and assign consumer
+	_, err = coordinator.CreateSubscription(owner)
+	require.NoError(t, err)
+	backend.Commit()
+	b, err := utils.GenericEncode([]string{"uint64"}, uint64(1))
+	require.NoError(t, err)
+	_, err = linkContract.TransferAndCall(owner, coordinatorAddress, big.NewInt(0), b)
+	require.NoError(t, err)
+	_, err = coordinator.AddConsumer(owner, 1, consumerAddress)
+	require.NoError(t, err)
+	_, err = consumer.SetSubscriptionID(random, 1)
+	require.Error(t, err)
+	_, err = consumer.SetSubscriptionID(owner, 1)
+	require.NoError(t, err)
+	_, err = consumer.RequestRandomWords(random)
+	require.Error(t, err)
+	_, err = consumer.RequestRandomWords(owner)
+	require.NoError(t, err)
+}
+
+func TestSimpleConsumerExample(t *testing.T) {
+	owner := newIdentity(t)
+	random := newIdentity(t)
+	genesisData := core.GenesisAlloc{
+		owner.From: {Balance: assets.Ether(10)},
+	}
+	backend := cltest.NewSimulatedBackend(t, genesisData, ethconfig.Defaults.Miner.GasCeil)
+	linkAddress, _, linkContract, err := link_token_interface.DeployLinkToken(
+		owner, backend)
+	require.NoError(t, err)
```

Comment:
Very minor point, but some whitespacing here to format into conceptual "paragraphs" would make this more readable